baybe/simulation/lookup.py

```diff
@@ -159,3 +139,41 @@ def _look_up_targets_from_dataframe(
 
     # Add the lookup values
     queries.loc[:, target_names] = np.asarray(all_match_vals)
+
+
+def label_columns(
+    input_labels: Sequence[str], output_labels: Sequence[str]
+) -> Callable:
+    """Create a decorator for labeling the inputs and outputs of array-based callables.
+
+    The decorator transforms a callable designed to work with unlabelled arrays such
+    that it can operate with dataframes instead. The original callable is expected to
+    accept and return two-dimensional arrays. When decorated, the callable accepts and
+    returns dataframes whose columns are mapped to the corresponding arrays based on the
+    specified label sequences.
+
+    Args:
+        input_labels: The sequence of input labels mapping the columns of the input
+            dataframe to columns of the input array in the specified order.
+        output_labels: The sequence of output labels mapping the columns of the output
+            dataframe to columns of the output array in the specified order.
+
+    Returns:
+        The decorator for the given input and output labels.
+    """
+
+    def decorator(
+        fn: Callable[[np.ndarray], np.ndarray],
+    ) -> Callable[[pd.DataFrame], pd.DataFrame]:
+        """Turn an array-based callable into a dataframe-based callable."""
+
+        @functools.wraps(fn)
+        def wrapper(df: pd.DataFrame, /) -> pd.DataFrame:
+            """Translate to/from an array-based callable using dataframes."""
+            array_in = df[input_labels].to_numpy()
+            array_out = fn(array_in)
+            return pd.DataFrame(array_out, columns=output_labels, index=df.index)
```

Comment:
should we add some validations taht ensure an good error message if there is a mismatch between arr and input_labels like a wrong or too smalls et has been provided?

Reply:
ok I wa shoping osme valdiationsa r epossible by just looking at the function definition onspection btu i guess thats not encessarily true

I donts ee the neeed for that class, why not simply do that as well in `wrapper`?

Reply:
I personally like the general idea of having a class based `Lookup` solution, but it also feels a bit overengineered.... let's be pragmatic here: If there is no easy solution for including meaningful error messages, let's go with the default ones. If people complain/we realize that the message are not helpful, we do another refactoring.

Reply:
I think that is the wrong place because we don't necessarily go via the wrapper. In fact, the wrapper is now only used with torch-based callables.

So let me suggest: I'll skip this part for now. The refactoring of the simulation module is anyway not yet over and we might still include it later, if we consider it helpful. But this way we postpone the decision until the overall picture of the remaining open changes is a bit clearer...

If you disagree, please reopen.